### ISSUE TITLE
Release RAT model checkpoints on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [CLAIRE-Labo/RAT#2]
Hi @wimh966 🤗

I'm Niels and work as part of the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2507.04416.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models for instance), you can also claim the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

Your work on RAT, bridging RNN efficiency and attention accuracy, looks really promising, especially with the reported speed improvements on 1.3B parameter models! I noticed your GitHub repository provides the code to train models like `rat-xl` and `rat_localattention_interleave-xl`, but I couldn't find direct download links for the pre-trained checkpoints themselves.

Would you be interested in hosting these pre-trained RAT model checkpoints you've developed on https://huggingface.co/models?
Hosting on Hugging Face will give you more visibility/enable better discoverability. We can add tags in the model cards so that people find the models easier, link it to the paper page, etc.

If you're down, leaving a guide [here](https://huggingface.co/docs/hub/models-uploading). If it's a custom PyTorch model, you can use the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin)
class which adds `from_pretrained` and `push_to_hub` to the model which lets you to upload the model and people to download and use models right away.
If you do not want this and directly want to upload model through UI or however you want, people can also use [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file).

After uploaded, we can also link the models to the paper page (read [here](https://huggingface.co/docs/hub/en/model-cards#linking-a-paper)) so people can discover your model.

You can also build a demo for your model on [Spaces](https://huggingface.co/spaces), we can provide you a ZeroGPU [grant](https://huggingface.co/docs/hub/en/spaces-gpus#community-gpu-grants), which gives you A100 GPUs for free.

Let me know if you're interested/need any guidance :)

Kind regards,

Niels

**Comment by wimh966** [comment]
Hi Niels,

Thanks a lot for your kind reminder and and very useful suggestion. Yes, this sounds very good to me. I'll complete it in December, ideally after NeurIPS. (sorry these days I've been super busy).

Once I'm done, I'll close the issue or I'll further contact you if I meet any problems and need your help during the process.

Thanks a lot for your help in advance! Have a nice day.

Best,
Xiuying

**Comment by wimh966** [comment]
Hi Niels,

Checkpoints have been pushed on https://huggingface.co/barpitf/RAT/tree/main. I kindly close the question. Feel free to reopen it if you have any question.

Best,
Xiuying

**Comment by NielsRogge** [comment]
Hey @wimh966, thanks for uploading the checkpoints and updating the README.

I noticed that the repository https://huggingface.co/barpitf/RAT appears to be private or not yet accessible (returns 404). Could you make it public so that others can access the checkpoints?

Once the repository is public, could you also link it to the paper page by mentioning the paper URL (https://huggingface.co/papers/2507.04416) in the model card? This will make the model automatically show up on the paper page.

See also here for more information: https://huggingface.co/docs/hub/model-release-checklist#write-a-comprehensive-model-card

Thanks

**Comment by wimh966** [comment]
Yes. I think it should work now!  Thank you! Have a nice day.

Best,
Xiuying
